src/robot_dart/graphics/camera_osr.hpp

```diff
@@ -46,35 +49,22 @@ namespace robot_dart {
         class CameraOSR : public BaseGraphics {
 
         public:
-            CameraOSR(const dart::simulation::WorldPtr& world, unsigned int width = 640, unsigned int height = 480, bool shadowed = true) : _world(world), _width(width), _height(height), _frame_counter(0), _enabled(true), _buffer_valid(true), _recording(false), _filename("camera_record.png"), _image(new osg::Image)
+            CameraOSR(const dart::simulation::WorldPtr& world, unsigned int width = 640, unsigned int height = 480, bool shadowed = true) : _world(world), _width(width), _height(height), _frame_counter(0), _enabled(true), _buffer_valid(true), _recording(false), _filename("camera_record.png"), _image(new osg::Image), _shot_done(false)
 
             {
+                if (!global::pbufferManager) {
+                    std::cout << "Error: pbufferManager seems not to be started" << std::endl;
+                    assert(0);
+                }
+
                 _osg_viewer = new dart::gui::osg::Viewer;
                 _osg_viewer->setThreadingModel(osgViewer::ViewerBase::ThreadingModel::SingleThreaded);
-                //graphics context
-                ::osg::ref_ptr<::osg::GraphicsContext::Traits> traits = new ::osg::GraphicsContext::Traits;
-                traits->readDISPLAY();
-
-                traits->x = 0;
-                traits->y = 0;
-                traits->width = width;
-                traits->height = height;
-                traits->red = 8;
-                traits->green = 8;
-                traits->blue = 8;
-                traits->alpha = 0;
-                //traits->depth = 24;
-                traits->windowDecoration = false;
-                traits->pbuffer = true;
-                traits->doubleBuffer = true;
-                traits->sharedContext = 0;
-                traits->setUndefinedScreenDetailsToDefaultScreen();
-
-                ::osg::ref_ptr<::osg::GraphicsContext> pbuffer = ::osg::GraphicsContext::createGraphicsContext(traits.get());
-
-                if (pbuffer.valid()) {
+                _osg_viewer->setUseConfigureAffinity(false);
```

Comment:
Didn't you say that this is available only for OSG>=3.6?

Reply:
Yes, it is and it is useful to assign the rendering threads properly. However, it should still run without this command. Maybe we can know the version of OSG installed and put this line in a conditional preprocessor command

Reply:
This was easy! `osg/Version`